### PR TITLE
Support node v4

### PR DESCRIPTION
**Pull request by wswoodruff**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -153,14 +159,32 @@ internals.initialize = function (server, next) {
...
+
+        knexConnections.push(rootCollector.knexGroups[key].knex);
+    });
+
+    // destroy() all knex instances
+    Items.parallel(knexConnections, (item, nxt) => {
+
+        item.destroy(() => {
+
+            nxt();
```
Comment (devinivy) [inline lib/index.js:181]: Not passing along errors on purpose?

File: lib/index.js [hunk 2]
```diff
@@ -153,14 +159,32 @@ internals.initialize = function (server, next) {
...
+        });
+    },
+    (_) => {
```
Comment (devinivy) [inline lib/index.js:184]: What's `_`?

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "description": "A hapi plugin integrating Objection.js ORM",
   "main": "lib/index.js",
   "scripts": {
-    "test": "NODE_ENV=test lab -a code -t 100 -L -I '__core-js_shared__'",
+    "test": "NODE_ENV=test lab --timeout 4000 -a code -t 100 -L -I '__core-js_shared__'",
```
Comment (devinivy) [inline package.json:7]: Which test is taking 4 seconds?

Comment (devinivy) [reply package.json:7]: As for `__core-js_shared__`, this is indeed an artifact of using babel.  We shouldn't need to use babel, so we oughtta figure out what's going on there.

File: lib/index.js (hunk 1 above)
Comment (wswoodruff) [reply lib/index.js:181]: It's weird, I was looking on their docs and Knex barely mentions destroy(). It seems it's only useful if you keep 'pools' open, which I don't involve in my tests. See here also no errors seem to be getting passed around if this is the correct destroy definition: https://github.com/tgriesser/knex/blob/a4bfee77513d000cf0fddf130a370071c240f93d/src/client.js#L267-L286

File: package.json (hunk 3 above)
Comment (devinivy) [reply package.json:7]: _Oh..._
Objection is probably compiled with babel.  Hmm...

File: lib/index.js (hunk 2 above)
Comment (wswoodruff) [reply lib/index.js:184]: That's an error, I wasn't sure how to cause an error to happen in order to cover it with tests so I copped out with the `_` any ideas?

Comment (devinivy) [reply lib/index.js:184]: No cop-outs ;) We don't want to bury any potential issues with the library.  You should be able to add an item to `rootCollector.knexGroups` that will cause an error.

Comment (wswoodruff) [reply lib/index.js:184]: I agree loL, I'm trying to figure out a real-world example for what might actually cause an error here, I really dunno

Comment (devinivy) [reply lib/index.js:184]: You don't need a real-world example, necessarily– you can just add a mock item to those knex-groups at runtime.

Comment (wswoodruff) [reply lib/index.js:184]: The destroy function (same link as above -->) https://github.com/tgriesser/knex/blob/a4bfee77513d000cf0fddf130a370071c240f93d/src/client.js#L267-L286 doesn't grab the reject part of the promise there on line 268. So is it possible to catch an error from the function? I can cause an exception but I don't know how to catch and handle an error with it

Comment (devinivy) [reply lib/index.js:184]: Looks like in practice it never fails.  But what if some day there is some sort of failure state?  Just pass `next` without wrapping it in an arrow– then we don't have to test whether any errors get passed along.  Let js and knex do the work!

Comment (wswoodruff) [reply lib/index.js:184]: Ok locally I have that stuff taken care of, just passing nxt to destroy(nxt);, but I won't be able to hit the lines checking for an error from there now so I can't have 100% coverage, wut do I doooo?

Comment (wswoodruff) [reply lib/index.js:184]: I created a PR for the changes. Here are the lines in question: https://github.com/BigRoomStudios/schwifty/blob/1b9620f92fd9da1787410c291bf819bef01a7f2d/lib/index.js#L192-L194

Comment (devinivy) [reply lib/index.js:184]: 1. You don't want to throw errors in async functions unless something completely unrecoverable has occurred.
2. You don't need to destroy (or consider destroying) the root knex connection twice.  Is there any reason that's done?  Is there any reason the root connection is destroyed first?

Keepin it simple!
```js
Items.parallel(knexConnections, (item, nxt) => item.destroy(nxt), next);
```

Comment (wswoodruff) [reply lib/index.js:184]: Awesome, thank! One-liner!

**Pull request merged**